clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -965,6 +965,10 @@ public void flush() {
     @Override
     public List<PartitionInfo> partitionsFor(String topic) {
         try {
+            if (topic == null) {
+                log.error("topic cannot be null");
```

Comment:
Hmm, there's no reason to log this. Also, the check should be outside the `try`. Something like the following would do:

```java
Objects.requireNonNull(topic, "topic cannot be null")
```

I'd use a similar pattern in `Metadata.add`.

Reply:
Thanks for the suggestion. much cleaner now. updated the PR.